### PR TITLE
WIP: Navigation

### PR DESCRIPTION
**Pull request by SyrianSpock**
Achieved:
- Navigation in simulation
- Navigation on robot but with navigation stack running on PC

To do:
- Try running navigation stack directly on robot
- Test static obstacle avoidance
- Tune planners and controllers

### CODE REVIEW

**Comment by antoinealb** [comment]
Any reason that the navigation stack cannot run on the BBB for now ?

Reading the code now.

**Comment by SyrianSpock** [comment]
I just didn't have the time.

File: goldorak_simulation/config/control.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  # to start
+  pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
+  twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
+
+  # Top level frame (link) of the robot description
+  base_frame_id: base_link
+
+  # Velocity and acceleration limits for the robot
+  linear:
+    x:
```
Comment (antoinealb) [inline goldorak_simulation/config/control.yaml:31]: What is the difference between this and the limits defined in `goldorak_navigation/param/base_local_planner_params.yaml` ? Is that one purely the simulation limit ?

Comment (SyrianSpock) [inline goldorak_simulation/config/control.yaml:31]: These are implemented as clamps on the simulated controller side.
It's not used by the "real controller" on the robot, although we could benefit from implementing clamps.

The others in `goldorak_navigation/param/base_local_planner_params.yaml` are limitations imposed on the planner that model the dynamics of the system more accurately: static friction is modeled with the minimum velocity applicable.

Comment (SyrianSpock) [inline goldorak_simulation/config/control.yaml:31]: Actually the "real controller" is not even a controller, it just converts velocity commands from the planner into wheel setpoints, so it's basically open-loop. Maybe we should implement a PID to follow the commands with better performance.

**Comment by SyrianSpock** [comment]
I think we should run this on hardware ASAP and merge.

**Comment by antoinealb** [comment]
agree. wanna try on wednesday ?

**Pull request merged**